Valet/VALValet.m

```diff
@@ -128,24 +128,46 @@ OSStatus VALAtomicSecItemDelete(CFDictionaryRef query)
 
 @interface VALValet ()
 
-/// Stores the root query to be used in all SecItem queries.
-@property (copy, readonly) NSDictionary *baseQuery;
-
 /// The service identifier within the baseQuery (kSecAttrService).
-@property (copy, readonly) NSString *secServiceIdentifier;
+@property (nonnull, copy, readonly) NSString *secServiceIdentifier;
 
 /// Set and Remove must be atomic operations relative to one another to ensure that SecItemUpdate is never called on an item that has been removed from the keychain.
-@property (copy, readonly) NSLock *lockForSetAndRemoveOperations;
+@property (nonnull, copy, readonly) NSLock *lockForSetAndRemoveOperations;
 
 @end
 
 
 @implementation VALValet
 
+#pragma mark - Protected Class Methods
+
++ (nullable NSMutableDictionary *)mutableBaseQueryWithIdentifier:(nonnull NSString *)identifier accessibility:(VALAccessibility)accessibility initializer:(nonnull SEL)initializer;
+{
+    VALCheckCondition(identifier.length > 0, nil, @"Must provide a valid identifier");
+    
+    return [@{
+              // Valet only handles generic passwords.
+              (__bridge id)kSecClass : (__bridge id)kSecClassGenericPassword,
+              // Use the identifier, Valet type and accessibility settings to create the keychain service name.
+              (__bridge id)kSecAttrService : [NSString stringWithFormat:@"VAL_%@_%@_%@_%@", NSStringFromClass(self), NSStringFromSelector(initializer), identifier, VALStringForAccessibility(accessibility)],
```

Comment:
As much as I'd like to see the initializer selector removed from this, I realize that with secure enclave we can't migrate the `kSecAttrService` so this is fine as is.